chain/types/signedmessage.go

```diff
@@ -78,7 +78,14 @@ func (sm *SignedMessage) MarshalJSON() ([]byte, error) {
 }
 
 func (sm *SignedMessage) ChainLength() int {
-	ser, err := sm.Serialize()
+	var ser []byte
+	var err error
+	if sm.Signature.Type == crypto.SigTypeBLS {
+		// BLS chain message length doesn't include signature
+		ser, err = sm.Message.Serialize()
+	} else {
+		ser, err = sm.Serialize()
+	}
```

Comment:
Looks like a fix for a different issue, mildly fishy consensus-wise, though looking at usages I believe it's fine

Reply:
Yeah, it is a fix for the mempool. Doesn't touch consensus as we have no signed BLS message in VM.